### ISSUE TITLE
getRoot typings in MST3

### ISSUE THREAD

**Issue by PMExtra** [mobxjs/mobx-state-tree#951]
I have:

* [x] I think something is not working as it should.
  * [x] I've checked documentation and searched for existing issues
  * [x] I've made sure your project is based on the latest MST version
  * [x] Fork [this](https://codesandbox.io/s/jlnmjqr7xv) code sandbox or another minimal reproduction. 
  * [x] Describe expected behavior
  * [x] Describe observed behavior

In MST2:
```
const RootStoreModel = types.props({ subStore: SubStore })
type RootStore = typeof RootStoreModel.Type

const SubStoreModel = types.actions(self => ({
  someFunc() {
    let rootStore = getRoot<RootStore>(self)
    // I got a variable `rootStore: RootStore`
  }
}))
```

In MST3:
```
    let rootStore = getRoot<RootStore>(self)
    // I got an error TS2344: Types 'blabla' does not satisfy the constraint 'IAnyType'
    // Then I got a variable `rootStore: ExtractIStateTreeNode<RootStore, ExtractC<RootStore>, ExtractS<RootStore>, ExtractT<RootStore>>`
    // And I cannot get auto completion or definition of RootStore in WebStorm.
```

Everything looks fine at runtime even the type was wrong.
What's wrong with MST3?

**Comment by xaviergonz** [comment]
do you have something named 'blabla' somewhere? if so, how does it look like?

**Comment by xaviergonz** [comment]
actually try getRoot<typeof RootStoreModel>

**Comment by xaviergonz** [comment]
I'll make later a PR so it supports both modes

**Comment by PMExtra** [comment]
@xaviergonz Thank you very much. So it's recommended to use `typeof RootStoreModel` instead `typeof RootStoreModel.Type`, right?

Now I have a template that like below:

SomeStoreModel.ts
```
    export const SomeStoreModel = types.blabla...
    export type SomeStore = typeof SomeStoreModel.Type
    export type SomeStoreSnapshot = typeof SomeStoreModel.SnapshotType
```

Should I replace all `typeof SomeStoreModel.Type` with `typeof SomeStoreModel` ?

**Comment by xaviergonz** [comment]
After the PR I created gets merged you will basically be able to use both ways and both ways should work.
Just bear in mind that for non-model types (e.g. maps and arrays) you have to use typeof Array/Map.

This is, after the PR all this will work
```
    const C = types.model({ a: 123 })

    // model as root
    const ModelWithC = types.model({ c: C })
    const modelInstance = ModelWithC.create({ c: C.create() })

    // getRoot
    const modelRoot1 = getRoot<typeof ModelWithC>(modelInstance.c)
    const modelCR1: typeof C.Type = modelRoot1.c
    const modelRoot2 = getRoot<typeof ModelWithC.Type>(modelInstance.c)
    const modelCR2: typeof C.Type = modelRoot2.c

    // getParent
    const modelParent1 = getParent<typeof ModelWithC>(modelInstance.c)
    const modelCP1: typeof ModelWithC.Type = modelParent1
    const modelParent2 = getParent<typeof ModelWithC.Type>(modelInstance.c)
    const modelCP2: typeof ModelWithC.Type = modelParent2

    // array as root
    const ArrayOfC = types.array(C)
    const arrayInstance = ArrayOfC.create([C.create()])

    // getRoot
    const arrayRoot1 = getRoot<typeof ArrayOfC>(arrayInstance[0])
    const arrayCR1: typeof C.Type = arrayRoot1[0]

    // getParent
    const arrayParent1 = getParent<typeof ArrayOfC>(arrayInstance[0])
    const arrayCP1: typeof ArrayOfC.Type = arrayParent1

    // map as root
    const MapOfC = types.map(C)
    const mapInstance = MapOfC.create({ a: C.create() })

    // getRoot
    const mapRoot1 = getRoot<typeof MapOfC>(mapInstance.get("a")!)
    const mapC1: typeof C.Type = mapRoot1.get("a")!

    // getParent
    const mapParent1 = getRoot<typeof MapOfC>(mapInstance.get("a")!)
    const mapCP1: typeof MapOfC.Type = mapParent1
```

**Comment by mweststrate** [comment]
Fix released as 3.1.1
